launch/launch/launcher.py

```diff
@@ -320,6 +391,9 @@ def __init__(self, launcher):
         super(AsynchronousLauncher, self).__init__()
         self.launcher = launcher
 
+    def terminate(self):
+        self.launcher.interrupt_launch()
```

Comment:
`terminate` does not wait on something to be done in my experience. For example, subprocess's terminate just sends the signal: https://docs.python.org/3.5/library/subprocess.html#subprocess.Popen.terminate

The expectation is that you need to then wait for the thing to finish, which is what `join` is for and I think it makes sense to keep them as separate steps. I use `join` in conjunction with this in the tests:

https://github.com/ros2/launch/pull/25/files#diff-b620208b20ed521101ca3dbcc3cdc2e1R43